include/swift/Remote/MetadataReader.h

```diff
@@ -838,8 +832,10 @@ class MetadataReader {
       if (!Result)
         return BuiltType();
 
-      auto flags = FunctionTypeFlags().withConvention(Function->getConvention())
-                                      .withThrows(Function->throws());
+      auto flags = FunctionTypeFlags()
+                       .withConvention(Function->getConvention())
+                       .withThrows(Function->throws())
+                       .withParameterFlags(Function->hasParameterFlags());
       auto BuiltFunction =
           Builder.createFunctionType(Parameters, Result, flags);
```

Comment:
I think you'll need to change the createFunctionType API to also take parameter flags, which probably means remapping them to the AST flags.

Also, what I mean about TupleMetadata is that you should modify the ``MetadataKind::Function`` case in ``readMetadata`` to start by reading the FunctionTypeMetadata and then use that to figure out the complete size to read, like the case for ``MetadataKind::Tuple`` does.

Reply:
Ah ok, I think I understand know, I thought you meant treat flags like labels are treated in case of tuple...

Regarding `createFunctionType` API - each of the elements in `Parameters` already has flags attached to it, so I think flags are covered that way?...

Reply:
Oh, yes, I see, that makes sense.